modules/util/tld-formatter/src/main/java/com/liferay/tld/formatter/TLDFormatter.java

```diff
@@ -102,6 +109,41 @@ public Set<String> getModifiedFileNames() {
 		return _modifiedFileNames;
 	}
 
+	private String _documentToString(Node node) throws Exception {
+		UnsyncByteArrayOutputStream unsyncByteArrayOutputStream =
+			new UnsyncByteArrayOutputStream();
+
+		OutputFormat outputFormat = OutputFormat.createPrettyPrint();
+
+		outputFormat.setExpandEmptyElements(false);
+		outputFormat.setIndent(StringPool.TAB);
+		outputFormat.setLineSeparator(StringPool.NEW_LINE);
+		outputFormat.setTrimText(true);
+
+		XMLWriter xmlWriter = new XMLWriter(
+			unsyncByteArrayOutputStream, outputFormat);
+
+		xmlWriter.write(node);
+
+		UnsyncBufferedReader unsyncBufferedReader = new UnsyncBufferedReader(
+			new UnsyncStringReader(
+				unsyncByteArrayOutputStream.toString(StringPool.UTF8)));
+
+		StringBundler sb = new StringBundler();
+
+		String line = null;
+
+		while ((line = unsyncBufferedReader.readLine()) != null) {
+			line = StringUtil.trimTrailing(line);
+
+			sb.append(line);
+
+			sb.append(StringPool.NEW_LINE);
+		}
```

Comment:
@tinatian Hi Tina,
The way we changed ServiceBuilder using below takes away the space before and after `<[[!CDATA  ]]>` because OutputFormat._padText is set to false

> 		OutputFormat outputFormat = new OutputFormat(StringPool.TAB, true);
>		outputFormat.setTrimText(true);
>		outputFormat.setOmitEncoding(true);

So, instead of using setPadText(false), I manually got rid of trailing white spaces.

below from the original Dom4jUtil logic also works too but it would iterate through entire content several times
> while (content.contains(" \n")) {
> 			content = StringUtil.replace(content, " \n", "\n");
> 		}